### ISSUE TITLE
win_spawn_process in process_windows.c.v does not work with paths that contain spaces

### ISSUE THREAD

**Issue by hungrybluedev** [vlang/v#18812]
### Describe the bug

I'm copying the discussions we have had on Discord here for archival purposes.

```
spytheman on Discord:

Process.win_spawn_process does:
    cmd := '${p.filename} ' + p.args.join(' ')
    cmd_wide_ptr := cmd.to_wide()
    to_be_freed << cmd_wide_ptr
    C.ExpandEnvironmentStringsW(cmd_wide_ptr, voidptr(&wdata.command_line[0]), 32768)
in vlib/os/process_windows.c.v:106
I think that needs fixing
with lots of tests; I'll check what can be done tomorrow, if someone does not fix it first before that.

...
it is blocking, and afaik vls and v-analyzer want to start vfmt without blocking, and monitor it themselves 
which is weird on its own, since vfmt is usually very fast, but 🤷‍♂️
```

TL;DR: We suspect that `Process.win_spawn_process` cannot work with paths that contain spaces and therefore, arguments must be quoted.

### Expected Behavior

V analyzer (and other programs on Windows that rely on os.Process) should be able to work on paths that contain spaces.

### Current Behavior

Programs on windows using `os.Process` cannot work with paths that contains spaces.

### Reproduction Steps

See discussion above.

### V version

V 0.4.0 e7e5a07

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.0 ab258ae.e7e5a07
OS: windows, Microsoft Windows 11 Home v22621 64-bit
Processor: 16 cpus, 64bit, little endian,

getwd: C:\Development\Repositories\VProjects\Typestates\typestate_v
vexe: C:\Development\Dependencies\v-main\v.exe
vexe mtime: 2023-07-07 12:44:24

vroot: OK, value: C:\Development\Dependencies\v-main
VMODULES: OK, value: C:\Development\Dependencies\vmodules
VTMP: OK, value: C:\Development\scratch\tmp

Git version: git version 2.41.0.windows.1
Git vroot status: weekly.2023.26-89-ge7e5a07a
.git/config present: true
```
